### PR TITLE
Fix: Update tabs to remain the same size between unselected and selected states

### LINKED ISSUE #23947: [Bug]: TabsList jumps pixels when you change tabs.... :(
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
chrome, windows
```

### Are you reporting Accessibility issue?

_No response_

### Reproduction

https://stackblitz.com/edit/react-ts-rtwqb7

tabs

## Actual Behavior
Change tabs, bold text changes cause pixel jumps around in the tab

## Expected Behavior
Change tabs, bold tabs doesn't add pixel width to a tab

This used to be a major eye sore in the Fabric Pivot and was later fixed (by always rendered "selected" text and hiding it so that bold widths would take layout space.)

Now it jumps around when you click around. Feels very unpolished.

Also of minor note is that the grey "hover underline" shows momentarily when you select a tab, while the brand colored underline gracefully glides on top of it. Recommend not showing the hover underline when something becomes selected... would make it much more polished.

### Logs

```shell
n/a
```

### Requested priority

Normal

### Products/sites affected

any using tabs

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by GeoffCoxMSFT**
## Fix
Currently tab size is based on the content within.  When a tab is selected, a bold font is used.  This causes tabs to be different sizes between unselected and selected states creating a visual movement of tabs.

This updates TabList and Tab to reserve space in unselected tabs to the content is the same size for both states.  This eliminates the visual movements of tabs.

## Changes
- Added reserveSelectedTabSpace to TabList context
- Added contentReservedSpaceClassName to TabState
- Updated useTab to set contentReservedSpaceClassName based on context reserveSelectedTabSpace
- Updated useTabStyles to set contentReservedSpaceClassName style to match selected tab content style
- Updated renderTab to render content slot again with with selected tab content style and visibility:hidden
- Updates test snapshots

## Issues
Fixes #23947

## Open issues
Need to measure perf impact of rendering content twice in every tab.

### Before
See associated issue.

### After
https://user-images.githubusercontent.com/28762486/200083589-b01f4332-6d06-4d6e-8f46-6e0e81ea7f3a.mov

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `62.941 kB`<br />`17.678 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.907 kB`<br />`52.983 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.446 kB`<br />`11.033 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/81a3171e14fcaead04c6cf3fbf8eda962178b442'>81a3171e14fcaead04c6cf3fbf8eda962178b442</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/Avatar-mount_base.html">1314</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/Avatar-mount.html">1314</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/Button-mount_base.html">960</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/Button-mount.html">940</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProvider-mount_base.html">1576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProvider-mount.html">1609</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-mount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-mount.html">632</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">591</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">600</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">621</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/MakeStyles-mount_base.html">1920</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/MakeStyles-mount.html">1907</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/SpinButton-mount_base.html">2522</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25542/perf-test-react-components/SpinButton-mount.html">2509</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-tabs/src/components/TabList/TabList.types.ts [hunk 1]
```diff
@@ -47,6 +47,15 @@ export type TabListProps = ComponentProps<TabListSlots> & {
    */
   appearance?: 'transparent' | 'subtle';
 
+  /**
+   * Tab size may change between unselected and selected states.
+   * The default scenario is a selected tab has bold text.
+   *
+   * When true, this property requests tabs be the same size whether unselected or selected.
+   * @default 'true'
```
Comment (micahgodbolt) [inline packages/react-components/react-tabs/src/components/TabList/TabList.types.ts:55]: default true is typically an antipattern. is there a specific need to do it this way?

Comment (spmonahan) [inline packages/react-components/react-tabs/src/components/TabList/TabList.types.ts:55]: super nit
```suggestion
   * @default true
```

Comment (GeoffCoxMSFT) [reply packages/react-components/react-tabs/src/components/TabList/TabList.types.ts]: A variable named in the the negative is generally regarded as confusing to callers (e.g. disableReserveSelectedTabSpace).  The default case here is to have the built-in Tab not pixel jump between unselected and selected states, so it seems the best approach to have the feature on by default.

**Review by spmonahan — APPROVED** [review]

**Review by micahgodbolt — APPROVED** [review]

**Pull request merged**